### PR TITLE
Add support for PCI DSS v3.2.1 for SLE12

### PR DESCRIPTION
**Pull request by teacup-on-rockingchair**
#### Description:

- Add PCI-DSS control for v.3.2.1 and SLE profiles to include it

#### Rationale:

- Add sle12 platform and missing CCE and PCIDSS references to rules related to PCIDSS effort for SLE platforms

### CODE REVIEW

**Comment by ggbecker** [comment]
I have realized that maybe we could have a single `pcidss_v3` that would contain both content for OCP4 and other products as well, since the controls file will include rules that are applicable to a given built product, so let's say you build a profile using this controls file, it will include only the rules that have `prodtype` set to that given product.

**Comment by teacup-on-rockingchair** [comment]
> I have realized that maybe we could have a single `pcidss_v3` that would contain both content for OCP4 and other products as well, since the controls file will include rules that are applicable to a given built product, so let's say you build a profile using this controls file, it will include only the rules that have `prodtype` set to that given product.

I thought about it also but what stopped me of reusing the existing control was not rules that were mostly OCP4 specific, but the notes in the pcidss_ocp4 control file which had a lot of references to Open Shift specific behaviour and OCP environment. If you think that it would be ok to put prodtype specific jinja clauses around those notes, then I guess we can easily aggregate both controls.

**Comment by ggbecker** [comment]
> > I have realized that maybe we could have a single `pcidss_v3` that would contain both content for OCP4 and other products as well, since the controls file will include rules that are applicable to a given built product, so let's say you build a profile using this controls file, it will include only the rules that have `prodtype` set to that given product.
> 
> I thought about it also but what stopped me of reusing the existing control was not rules that were mostly OCP4 specific, but the notes in the pcidss_ocp4 control file which had a lot of references to Open Shift specific behaviour and OCP environment. If you think that it would be ok to put prodtype specific jinja clauses around those notes, then I guess we can easily aggregate both controls.

It's probably ok to have a separate file then. There is also the fact that it can create a misinterpretation of the `status` field depending on which rules are applicable to which product. So let's keep them separate, at least for now. Then this `pcidss_3` can be used for the general linux distros.

File: products/sle12/profiles/pci-dss.profile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+documentation_complete: true
+
+metadata:
+    SMEs:
+        - abergmann
+
+reference: https://docs-prv.pcisecuritystandards.org/PCI%20DSS/Standard/PCI-DSS-v4_0.pdf
```
Comment (ggbecker) [inline products/sle12/profiles/pci-dss.profile:7]: 3.2.1 instead?

**Review by ggbecker — COMMENTED** [review]
Hi, I've run the profile diff tool and notice that the following rules that were previously in the pci-dss.profile are not present in the pci-dss controls file:

`$ ./build-scripts/profile_tool.py sub --profile1 products/sle15/profiles/pci-dss-from-main-branch.profile --profile2 build/sle15/profiles/pci-dss.profile --product sle15 --build-config-yaml build/build_config.yml --ssg-root .`

```
selections:
- accounts_minimum_age_login_defs
- accounts_no_uid_except_zero
- accounts_password_warn_age_login_defs
- accounts_tmout
- accounts_umask_etc_bashrc
- accounts_umask_etc_login_defs
- accounts_umask_etc_profile
- cracklib_accounts_password_pam_dcredit
- cracklib_accounts_password_pam_lcredit
- cracklib_accounts_password_pam_minlen
- cracklib_accounts_password_pam_ocredit
- cracklib_accounts_password_pam_retry
- cracklib_accounts_password_pam_ucredit
- file_permissions_sshd_private_key
- file_permissions_sshd_pub_key
- no_direct_root_logins
- package_audit_installed
- package_chrony_installed
- package_openldap-clients_removed
- package_sudo_installed
- package_telnet-server_removed
- package_vsftpd_removed
- package_ypserv_removed
- postfix_network_listening_disabled
- securetty_root_login_console_only
- sshd_disable_empty_passwords
- sshd_disable_root_login
- sshd_do_not_permit_user_env
- sshd_enable_warning_banner
- sshd_set_loglevel_verbose
- sudo_add_use_pty
- sudo_custom_logfile
- var_smartcard_drivers=cac
```

Maybe you would like to include these rule as well so you don't end up with a different profile, at least for the transition part.

File: products/sle12/profiles/pci-dss.profile (hunk 1 above)
Comment (teacup-on-rockingchair) [reply products/sle12/profiles/pci-dss.profile]: Should be ok in [0e0c5db](https://github.com/ComplianceAsCode/content/pull/9613/commits/0e0c5db0fe96c5e52f8bf788041cd8350d629baa)

**Comment by teacup-on-rockingchair** [comment]
> 



> Hi, I've run the profile diff tool and notice that the following rules that were previously in the pci-dss.profile are not present in the pci-dss controls file:
> 
> `$ ./build-scripts/profile_tool.py sub --profile1 products/sle15/profiles/pci-dss-from-main-branch.profile --profile2 build/sle15/profiles/pci-dss.profile --product sle15 --build-config-yaml build/build_config.yml --ssg-root .`
> 
> ```
> selections:
> - accounts_minimum_age_login_defs
> - accounts_no_uid_except_zero
> - accounts_password_warn_age_login_defs
> - accounts_tmout
> - accounts_umask_etc_bashrc
> - accounts_umask_etc_login_defs
> - accounts_umask_etc_profile
> - cracklib_accounts_password_pam_dcredit
> - cracklib_accounts_password_pam_lcredit
> - cracklib_accounts_password_pam_minlen
> - cracklib_accounts_password_pam_ocredit
> - cracklib_accounts_password_pam_retry
> - cracklib_accounts_password_pam_ucredit
> - file_permissions_sshd_private_key
> - file_permissions_sshd_pub_key
> - no_direct_root_logins
> - package_audit_installed
> - package_chrony_installed
> - package_openldap-clients_removed
> - package_sudo_installed
> - package_telnet-server_removed
> - package_vsftpd_removed
> - package_ypserv_removed
> - postfix_network_listening_disabled
> - securetty_root_login_console_only
> - sshd_disable_empty_passwords
> - sshd_disable_root_login
> - sshd_do_not_permit_user_env
> - sshd_enable_warning_banner
> - sshd_set_loglevel_verbose
> - sudo_add_use_pty
> - sudo_custom_logfile
> - var_smartcard_drivers=cac
> ```
> 
> Maybe you would like to include these rule as well so you don't end up with a different profile, at least for the transition part.

Those rules were added in the context of PCIDSS v4 effort, thanks for the feedback. For now I think is best if I include them in the context of different profile file. Once I have built a control file for that version of the standard will include them and rework the profile file.

**Comment by ggbecker** [comment]
We should probably shift towards having version 4 as the default one. This basically means that the pci-dss.profile should point to the latest 4 version. Version 3 should eventually go away when there is the sunset and then we are left only with the pci-dss.profile identifier for example. The same for the `pcidss` reference identifier. We should update the URL links to the latest version and create a pcidss_3 reference URI to point to the older policy. But one problem might be that the already assigned references will be inconsistent with the v4 for example. What do you think?

File: products/sle12/profiles/pci-dss.profile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+documentation_complete: true
+
+metadata:
+    SMEs:
+        - abergmann
+
+reference: https://www.pcisecuritystandards.org/documents/PCI_DSS_v3-2-1.pdf
+
+title: 'PCI-DSS v3.2.1 Control Baseline for SUSE Linux enterprise 12 and 15'
```
Comment (ggbecker) [inline products/sle12/profiles/pci-dss.profile:9]:
```suggestion
title: 'PCI-DSS v3.2.1 Control Baseline for SUSE Linux enterprise 12'
```

File: products/sle12/profiles/pci-dss.profile [hunk 3]
```diff
@@ -0,0 +1,16 @@
+documentation_complete: true
+
+metadata:
+    SMEs:
+        - abergmann
+
+reference: https://www.pcisecuritystandards.org/documents/PCI_DSS_v3-2-1.pdf
```
Comment (ggbecker) [inline products/sle12/profiles/pci-dss.profile:7]: Something like:
```suggestion
reference: https://docs-prv.pcisecuritystandards.org/PCI%20DSS/Standard/PCI-DSS-v3-2-1-r1.pdf
```

**Comment by teacup-on-rockingchair** [comment]
> We should probably shift towards having version 4 as the default one. This basically means that the pci-dss.profile should point to the latest 4 version. Version 3 should eventually go away when there is the sunset and then we are left only with the pci-dss.profile identifier for example. The same for the `pcidss` reference identifier. We should update the URL links to the latest version and create a pcidss_3 reference URI to point to the older policy. But one problem might be that the already assigned references will be inconsistent with the v4 for example. What do you think?

In general I agree, but would prefer to do that, in the context of another PR, once I generate the control file for v4 and then will be able to rename the profiles per product. Because the v4 for SLE platforms is WIP, and I am not really aware the state of the art for the other platforms.

**Comment by teacup-on-rockingchair** [comment]
> Also, if you could please rebase on top of latest main branch to check if there is no conflict with the CCE assignment since rumch-se has been including a bunch of them, just to be sure that the test will be able to catch any duplicate. Thanks. Otherwise I think we can probably merge this one.

Thanks for the reminder, I just did the rebase and run the uniqe and missing cce tests for sle platforms.

**Review by ggbecker — APPROVED** [review]

**Pull request merged**